pkgs/README.md

```diff
@@ -23,7 +23,20 @@ See the [CONTRIBUTING.md](../CONTRIBUTING.md) document for more general informat
 
 ## Quick Start to Adding a Package
 
-To add a package to Nixpkgs:
+Each new package has a cost for the maintainers, Hydra and users downloading
+Nixpkgs. Before adding a new package, please consider the following questions:
+* Is the package ready for general use? For example, if the author(s) is/are working towards the first release, it is probably too early.
```

Comment:
Remark: I will go further, I had discussions with @Foxboron on this who had some of his projects "too early packaged" and he was not really comfortable with this.

For example, we can also be explicit and say "if the author explicitly asks for not being packaged", etc. I mean, it's obvious in retrospective in packaging `unstable-XYZ`-type of packages.

I would just like to distinguish:

- software that is used and can be used because it has been there for a while, but no release or security fixes in latest HEAD, etc.
- software that is being actively developed and can be already ~usable as a sneak peek, but with absolute no quality guarantees.

The former is fine without "first release", the latter is very much meh IMHO.

Reply:
The precedent is that we don't do much deference to upstream, though. (As in, multipled upstreams tried to stop people from packaging their announced-as-stable OSI-and-DSFG-approved-license software, and Nixpkgs ignored the requests)

Reply:
There is a difference between a maintainer going "this isn't stable software and I would urge you to wait" vs "I dont want my FOSS code published in a software repository".

I would ignore the latter if I didn't care about downstream support when encountering bugs.

Reply:
Ok, I addressed this in d5f8b87b2ece652c722a0091c793465e005f6406 . I tried compressing the decision tree so it doesn't become a full D&D adventure.